src/seedu/addressbook/util/FilePathUtil.java

```diff
@@ -0,0 +1,50 @@
+package seedu.addressbook.util;
+
+import java.io.File;
+import java.io.IOException;
+
+/**
+ * This is a utility class for file path validation.
+ */
+public class FilePathUtil {
+    private static final int EXTENSION_INVALID_INDEX = 0;
+
+    /**
+     * Returns true if the given file path is acceptable.
+     * The file path is acceptable if it has a folder that exists,
+     * has a file name that is acceptable to the operating system eg. does not contain <>:"/|?*,
+     * File name must also contain an extentsion
+     */
+    public static boolean isValidFilePath(String filePath) {
+        if (filePath == null) {
+            return false;
+        }
+        File filePathToValidate = new File(filePath);
+        if (!hasValidFileName(filePathToValidate)) {
+            return false;
+        }
+        return hasValidFolder(filePathToValidate);
+    }
+    
+    /**
+     * Returns true if the file path has a folder that exists
+     */
+    private static boolean hasValidFolder(File filePath) {
+        File folderToValidate = filePath.getAbsoluteFile().getParentFile();
+        return folderToValidate.exists();
+    }
+    
+    /**
+     * Returns true if file path has a valid file name.
+     * File name is valid if it has an extension and no reserved characters <>:"/|?*.
```

Comment:
Reserved characters are unfortunately platform-dependent though. e.g. for Windows it is [all the characters in red in this image](https://lh3.googleusercontent.com/0TTN3uK7Ha9GuS_3DP05qQSz15UQ1sYgEFh1DeJPZtvWwang1t1D3nW2zU-RndEFpGME-cy-UA53i-Vd8EXHBX_u1QLHP0BoqVq_9wWMhC55bFviWJHdDyaYtxzeD60O77AKNBHo). For other unix-like operating systems, AFAIK it is just '/' and '\0', but their underlying filesystems may impose further restrictions as well :-P Then of course there are other platforms which this project may wish to support in the future ;)

So maybe we should not bother trying to list all possible reserved characters.